Gems/Camera/Code/Source/CameraEditorSystemComponent.cpp

```diff
@@ -99,19 +102,40 @@ namespace Camera
 
         AzToolsFramework::ScopedUndoBatch undoBatch("Create Camera Entity");
 
-        // Create new entity
+        bool prefabSystemEnabled = false;
+        AzFramework::ApplicationRequests::Bus::BroadcastResult(
+            prefabSystemEnabled, &AzFramework::ApplicationRequests::IsPrefabSystemEnabled);
+
         AZ::EntityId newEntityId;
+
         AZ::EBusAggregateResults<AZ::EntityId> cameras;
         Camera::CameraBus::BroadcastResult(cameras, &CameraBus::Events::GetCameras);
         AZStd::string newCameraName = AZStd::string::format("Camera%zu", cameras.values.size() + 1);
-        AzToolsFramework::EditorEntityContextRequestBus::BroadcastResult(
-            newEntityId, &AzToolsFramework::EditorEntityContextRequests::CreateNewEditorEntity, newCameraName.c_str());
+
+        const auto worldFromView = AzFramework::CameraTransform(cameraState);
+        // Create new entity
+        if (!prefabSystemEnabled)
+        {
+            AzToolsFramework::EditorEntityContextRequestBus::BroadcastResult(
+                newEntityId, &AzToolsFramework::EditorEntityContextRequests::CreateNewEditorEntity, newCameraName.c_str());
```

Comment:
If this is not safe to call with the prefab system enabled, it potentially might be better to move the logic you have here _inside_ `CreateNewEditorEntity` (and have it just call `prefabIntegrationInterface->CreateNewEntityAtPosition`). It seems this could be a problem in more places than just the CameraComponent and we don't want this code repeated in lots of different places. Therefore it might be better to just move this inside `CreateNewEditorEntity` (or make a new call that does this `if` check and then deprecate the existing `CreateNewEditorEntity` if it should no longer be used and then replace other call sites with the new code.

Reply:
Actually, CreateNewEditorEntity is setup to work with the prefab system because EditorEntityOwnershipService calls PrefabEditorEntityOwnershipService to create the entity. In fact, that's why creating the new camera entity works in the first place. But it looks like there's a bug with undoing the action and undo is getting called twice. I tried an alternate approach to fix this that's in this PR : https://github.com/o3de/o3de/pull/6738 that avoids calling prefab classes directly from EditorEntityContext. @AMZN-Igarri , can you try the solution given in the above PR to see if that fixes the undo problem?

Reply:
Thanks very much for the feedback @srikappa-amzn, much appreciated 👍

Reply:
@srikappa-amzn Just tested it and it fixes the undo issue perfectly! Thanks for looking into this

Reply:
I wonder is there a way we could make a test for this? 😁 (as a follow-up PR as this is fixing a critical bug, but it would be incredibly useful to try and catch this from regressing - are they any examples @AMZN-Igarri could follow @srikappa-amzn ? Thanks!)

Reply:
I do see some tests in EditorEntityContextComponentTests for creating editor entities but it looks like undo actions are not tested there. Fetching the undo stack using 
`AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(m_undoStack, &AzToolsFramework::ToolsApplicationRequestBus::Events::GetUndoStack);`
and triggering undo should help testing undo actions. PrefabTestFixture.cpp and several other fixtures contains examples of this if references are needed

Reply:
Ah perfect! :D Perhaps after https://github.com/o3de/o3de/pull/6738 is merged @AMZN-Igarri could add such a test as it feels like it would be useful (this PR can then be closed). Thanks!